commands/build.go

```diff
@@ -412,12 +416,40 @@ func runBuild(ctx context.Context, dockerCli command.Cli, options buildOptions)
 }
 
 // getImageID returns the image ID - the digest of the image config
-func getImageID(resp map[string]string) string {
-	dgst := resp[exptypes.ExporterImageDigestKey]
-	if v, ok := resp[exptypes.ExporterImageConfigDigestKey]; ok {
-		dgst = v
+func getImageID(resp map[string]string, dockerUsingContainerdSnapshotter bool) string {
+	if dockerUsingContainerdSnapshotter {
+		// If Docker is using the containerd snapshotter, use the image digest, not
+		// the image config digest. See https://github.com/moby/moby/issues/45458.
+		return resp[exptypes.ExporterImageDigestKey]
+	}
+
+	return resp[exptypes.ExporterImageConfigDigestKey]
+}
+
+func isDockerUsingContainerdSnapshotter(ctx context.Context, dockerCli command.Cli, exports []*controllerapi.ExportEntry) bool {
+	usingContainerdSnapshotter := false
+
+	// If the build command has "-o type=docker,context=<mycontext>", then pass that
+	// context to docker.Features() below, so we can find out if the associated
+	// Docker engine is using the containerd snapshotter.
+	dockerCtx := ""
+	for _, e := range exports {
+		if e.Type == "docker" {
+			for k, v := range e.Attrs {
+				if k == "context" {
+					dockerCtx = v
+				}
+			}
+		}
 	}
-	return dgst
+
+	docker := dockerutil.NewClient(dockerCli)
+	features := docker.Features(ctx, dockerCtx)
+	if features[dockerutil.OCIImporter] {
+		usingContainerdSnapshotter = true
+	}
```

Comment:
I think this is best to check worker info using buildkit client for this entitlement similar to https://github.com/docker/buildx/blob/9730a20f6b0b1f15a3b917e90c83f8b97a62dfd2/driver/docker/driver.go#L81-L90